edge-modules/MetricsCollector/LogAnalyticsMetricsSync.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics;
+    using Microsoft.Extensions.Logging;
+
+    class LogAnalyticsMetricsSync : IMetricsSync
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("MetricsCollector");
+        readonly MessageFormatter messageFormatter;
+        readonly Scraper scraper;
+
+        public LogAnalyticsMetricsSync(MessageFormatter messageFormatter, Scraper scraper)
+        {
+            this.messageFormatter = Preconditions.CheckNotNull(messageFormatter);
+            this.scraper = Preconditions.CheckNotNull(scraper);
+        }
+
+        public async Task ScrapeAndSync()
+        {
+            try
+            {
+                IEnumerable<string> scrapedMetrics = await this.scraper.Scrape();
+                foreach (string scrape in scrapedMetrics)
+                {
+                    string workspaceId = Settings.Current.AzMonWorkspaceId;
+                    string workspaceKey = Settings.Current.AzMonWorkspaceKey;
+                    string logType = Settings.Current.AzMonLogType;
+                    AzureLogAnalytics.Instance.PostAsync(workspaceId, workspaceKey, this.messageFormatter.BuildJSON(scrape), logType);
```

Comment:
1 - This will be helpful, so I'll find one.

2 - Assuming you are talking about JSON version for Log Analytics Post() body. Is this something formally defined by log analytics? If so, it would be helpful.

3 - I thought it was an anti-pattern to await things that don't need to be awaited?

Reply:
1. See my other comment. I think it is that.

2. I'm thinking if there is a version associate with `this.messageFormatter.BuildJSON(scrape)`. It looks like the answer is yes, he has a `schemaVersion` to track which Json format.

3. After talking with Damon, if you don't `await` it would be just fired and forget about it. But you want to send it in order you want to do `await`.  He also has a concern about how the `PostAsync()` is used in `foreach()` which means you actually send out a bunch of POST request.  Would it be better to formulate all the json then send a POST request once?  (Also my bad on the `public async void PostAsync(string workspaceId, string sharedKey, string content, string logType)` should really be `public async Task PostAsync(string workspaceId, string sharedKey, string content, string logType)` Do you want me to fix this?

Reply:
1 - Sounds good

2 - I'll investigate

3 - Awaiting the call would provide consistent ordering for log analytics, so I'll change. I'll also make the change from void to task. For the post requests, you'll only be making a few post requests (one for each endpoint you are scraping, which is by default 1), so I think it is fine to send multiple. Sending multiple requests also allows you to send in the case where BuildJson fails for one or more endpoints. I'm going to modify the try / catch logic to support this.